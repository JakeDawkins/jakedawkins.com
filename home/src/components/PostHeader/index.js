import React from "react";
import "../Header/header.scss";

//given unix timestamp
const renderDate = (dateTime) => {
  let date = new Date(dateTime*1000); //to ms
  let dateString = date.toDateString();
  return <time itemProp="datePublished" dateTime={dateTime} className="date">{dateString}</time>
}

const PostHeader = (props) => (
  <header className="header-post" role="banner">
      <div className="content">
          {renderDate(props.dateTime)}
          <h1 className="post-title" itemProp="name">{props.title}</h1>
          <p itemProp="description" className="subtitle">{props.description}</p>
      </div>
       <a className="down" data-scroll href="#scroll"><svg className="icon icon-angle-down"><use xlinkHref="#icon-angle-down"></use></svg></a>
  </header>
);

export default PostHeader;
