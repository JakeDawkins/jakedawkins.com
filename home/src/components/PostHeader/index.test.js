import PostHeader from './';
import React from 'react';
import renderer from 'react-test-renderer';

describe("PostHeader", () => {

  it("should exist", () => {
    const postHeader = renderer.create(
      <PostHeader />
    );

    expect(postHeader).toMatchSnapshot();
  });

  it("should render props correctly", () => {
    // const postHeader = renderer.create(
    //   <PostHeader />
    // );

    const postHeaderWithProps = renderer.create(
      <PostHeader
        dateTime="12345"
        title="test title"
        description="description here"
      />
    );

    expect(postHeaderWithProps).toMatchSnapshot();
    // expect(postHeaderWithProps).not.toBe(postHeader);
  });

});
