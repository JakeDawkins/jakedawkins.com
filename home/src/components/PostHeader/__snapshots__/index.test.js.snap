exports[`PostHeader should exist 1`] = `
<header
  className="header-post header-site"
  role="banner">
  <div
    className="content">
    <h1
      className="post-title"
      itemProp="name" />
    <p
      className="subtitle"
      itemProp="description" />
  </div>
  <a
    className="down"
    data-scroll={true}
    href="#scroll">
    <svg
      className="icon icon-angle-down">
      <use
        xlinkHref="#icon-angle-down" />
    </svg>
  </a>
  <div>
    <div
      className="search-wrapper">
      <div
        className="search-form">
        <input
          className="search-field"
          placeholder="Search..."
          type="text" />
        <svg
          className="icon-remove-sign">
          <use
            xlinkHref="#icon-close" />
        </svg>
        <ul
          className="search-results search-list" />
      </div>
    </div>
    <div
      className="overlay"
      id="fade"
      onClick={[Function bound toggleMenu]} />
    <a
      className="slideButton fade"
      id="slide">
      <svg
        className="icon-menu"
        id="open"
        onClick={[Function bound toggleMenu]}
        style={
          Object {
            "display": "inline"
          }
        }>
        <use
          xlinkHref="#icon-menu" />
      </svg>
      <svg
        className="icon-menu"
        id="close"
        onClick={[Function bound toggleMenu]}
        style={
          Object {
            "display": "none"
          }
        }>
        <use
          xlinkHref="#icon-close" />
      </svg>
    </a>
    <aside
      className=""
      id="sidebar">
      <nav
        id="navigation">
        <h2>
          MENU
        </h2>
        <ul>
          <li>
            <a
              onClick={[Function bound handleClick]}
              style={Object {}}>
              Posts
            </a>
          </li>
          <li>
            <a
              onClick={[Function bound handleClick]}
              style={Object {}}>
              About
            </a>
          </li>
        </ul>
      </nav>
    </aside>
  </div>
</header>
`;

exports[`PostHeader should render props correctly 1`] = `
<header
  className="header-post header-site"
  role="banner">
  <div
    className="content">
    <time
      className="date"
      dateTime="12345"
      itemProp="datePublished">
      12 - 31 - 1969
    </time>
    <h1
      className="post-title"
      itemProp="name">
      test title
    </h1>
    <p
      className="subtitle"
      itemProp="description">
      description here
    </p>
  </div>
  <a
    className="down"
    data-scroll={true}
    href="#scroll">
    <svg
      className="icon icon-angle-down">
      <use
        xlinkHref="#icon-angle-down" />
    </svg>
  </a>
  <div>
    <div
      className="search-wrapper">
      <div
        className="search-form">
        <input
          className="search-field"
          placeholder="Search..."
          type="text" />
        <svg
          className="icon-remove-sign">
          <use
            xlinkHref="#icon-close" />
        </svg>
        <ul
          className="search-results search-list" />
      </div>
    </div>
    <div
      className="overlay"
      id="fade"
      onClick={[Function bound toggleMenu]} />
    <a
      className="slideButton fade"
      id="slide">
      <svg
        className="icon-menu"
        id="open"
        onClick={[Function bound toggleMenu]}
        style={
          Object {
            "display": "inline"
          }
        }>
        <use
          xlinkHref="#icon-menu" />
      </svg>
      <svg
        className="icon-menu"
        id="close"
        onClick={[Function bound toggleMenu]}
        style={
          Object {
            "display": "none"
          }
        }>
        <use
          xlinkHref="#icon-close" />
      </svg>
    </a>
    <aside
      className=""
      id="sidebar">
      <nav
        id="navigation">
        <h2>
          MENU
        </h2>
        <ul>
          <li>
            <a
              onClick={[Function bound handleClick]}
              style={Object {}}>
              Posts
            </a>
          </li>
          <li>
            <a
              onClick={[Function bound handleClick]}
              style={Object {}}>
              About
            </a>
          </li>
        </ul>
      </nav>
    </aside>
  </div>
</header>
`;
